test/ui/FacetSettingsTest.ts

```diff
@@ -129,10 +129,39 @@ export function FacetSettingsTest() {
       facetSettings = new FacetSettings(['alphaascending', 'alphadescending', 'score'], facet);
       facetSettings.build();
       facetSettings.open();
-      let ascendingSection = $$(facetSettings.settingsPopup).find('.coveo-facet-settings-item[data-direction="ascending"]');
+      const ascendingSection = $$(facetSettings.settingsPopup).find('.coveo-facet-settings-item[data-direction="ascending"]');
       expect($$(ascendingSection).hasClass('coveo-selected')).toBe(true);
       $$(facetSettings.getSortItem('score')).trigger('click');
       expect($$(ascendingSection).hasClass('coveo-selected')).toBe(false);
     });
+
+    describe('when closing the popup', () => {
+      beforeEach(() => {
+        facetSettings = new FacetSettings(['foo', 'bar'], facet);
+        spyOn(facetSettings, 'close');
+        facetSettings.build();
+        facetSettings.open();
+      });
+
+      it('should close the after a delay on mouseleave', done => {
+        $$(facetSettings.button).trigger('mouseleave');
+        setTimeout(() => {
```

Comment:
In jest there are some nice methods like `useFakeTimers` and `runAllTimers` that fast-forward time avoiding the need for setTimeouts in tests. It seems jasmine has something similar with `jasmine.clock().tick(<number of ms to fast-forward>)`. I think using `clock` is better practice.

https://makandracards.com/makandra/32477-testing-settimeout-and-setinterval-with-jasmine
https://stackoverflow.com/questions/10955201/how-to-test-a-function-which-has-a-settimeout-with-jasmine

Reply:
TIL !

Much better, the tests run faster.